developer-docs-site/docs/guides/nfts/mint-nft-v2-cli.md

```diff
@@ -0,0 +1,918 @@
+intentional in some cases, but should be considered before hand. Always be highly aware of how you grant access to a resource account's signer capability.
+:::
+
+### Publishing the module and running the contract
+
+Publishing the module is basically the same as before. Just make sure you're in the `2-Using-Resource-Account` directory and run this command, note the only thing that changed is the module name in the first line, `create_nft_with_resource_account` instead of `create_nft`:
+
+```shell
+aptos move publish --named-addresses mint_nft_v2_part2=default --profile default --assume-yes
+```
+
+Call this function as the owner of the contract, which is our `default` profile. Keep in mind the `--profile default` flag:
+
+```shell
+aptos move run --function-id default::create_nft_with_resource_account::initialize_collection   \
+               --profile default                                          \
+               --args                                                     \
+                  string:"Krazy Kangaroos"                                \
+                  string:"https://www.link-to-your-collection-image.com"  \
+                  u64:3                                                   \
+                  u64:5                                                   \
+                  u64:100                                                 \
+                  string:"Krazy Kangaroo #1"                              \
+                  string:"https://www.link-to-your-token-image.com"       
+```
+
+Next we need to get the resource address for the contract with our view function.
+
+```shell
+aptos move view --function-id default::create_nft_with_resource_account::get_resource_address \
+                --profile default \
+                --args string:"Krazy Kangaroos"
+```
+
+Now we call this function as a user, which we simulate with our `nft-receiver` profile:
+
+```shell
+aptos move run --function-id default::create_nft_with_resource_account::mint \
+               --profile nft-receiver \
+               --args address:YOUR_RESOURCE_ADDRESS_HERE
+```
+
+Great! Now you've created the collection as an owner and requested to mint as a user and received the newly minted NFT.
+
+It may not feel different since you're acting as the owner and the receiver all from the command line, but in an actual dapp this user flow makes much more sense than before.
+
+In the first section, the user has to wait for the owner of the contract to mint and send them an NFT. In the second section, the user can request to mint and receive an NFT themselves.
+
+## 3. Adding restrictions: a whitelist, an end time, an admin, and an enabled flag
+
+We're still missing some very common features for NFT minting contracts:
+
+1. A whitelist that restricts minting to whitelisted addresses
+2. The ability to add/remove addresses from the whitelist
+3. A start* and end time
```

Comment:
Why not just say "An end time" then if start time is not being added :)

Reply:
I was trying to clarify that a start time is usually needed, but that we weren't adding one yet, lol. I'll remove, it's kinda pointless to have